gio/src/task.rs

```diff
@@ -59,11 +60,58 @@ impl Task {
             ffi::g_task_set_priority(self.to_glib_none().0, priority.into_glib());
         }
     }
+}
+
+pub trait TaskExtManual<T: ValueType> {
+    #[doc(alias = "g_task_return_value")]
+    fn return_value(&self, result: &T);
+
+    #[doc(alias = "g_task_propagate_value")]
+    fn propagate_value(&self) -> Result<T, glib::Error>;
```

Comment:
Are these two functions thread-safe? E.g. can you call safely `return_value()` from the original thread where the `Task` was created and inside the `run_in_thread()` closure at the same time?

Reply:
The answer is: no, for both functions. That's yet another problem with this API then.